napalm/base/base.py

```diff
@@ -218,9 +218,33 @@ def compare_config(self):
         """
         raise NotImplementedError
 
-    def commit_config(self, message=""):
+    def commit_config(self, message="", confirm_timeout=None):
         """
         Commits the changes requested by the method load_replace_candidate or load_merge_candidate.
+
+        :param message: Optional - configuration session commit message
+        :type message: str
+        :param confirm_timeout: Optional - The # of seconds before the configuration must be
+        confirmed via the commit_confirm method before OS reverts back to the original configuration
+        :type confirm_timeout: int
+        """
+        raise NotImplementedError
+
+    def commit_confirm(self):
+        """
+        Confirm the changes requested via commit_config if a confirm_timeout is set
+        """
+        raise NotImplementedError
+
+    def commit_confirm_revert(self):
```

Comment:
Still, I'm not sure if understand the distinction. 

_"1. Just rollback"_ by definition would revert the configuration to the previous state, no matter if there's an ongoing `commit_confirm` operation or not. As such, once you perform _(1)_, there should not be any commit confirm operation to cancel _(2)_ because the change was already reverted.
How _(1)_ is implemented to achieve the same results in all drivers it's a different topic

Reply:
On some platforms a rollback would not necessarily cancel a pending `commit confirm`. In other words, on some platforms we are artificially creating these so even though we have executed `rollback`, there still would be a pending `commit confirm` that would show up.

I am pretty sure this would be the case on Cisco IOS.

Also I want to avoid conflating the `commit_confirm` operations with the non-commit confirm operations.

Reply:
I'm going to assume we're all good here and sticking with the reversion methods for now.